### PR TITLE
Initial Build

### PR DESCRIPTION
**Pull request by mbilokonsky**
See readme for details. Reach out to me if you want help configuring your slack app, there's a lot of little knobs and buttons to tweak to make this work.

### CODE REVIEW

File: lib/behavior/greeting/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module.exports = controller => {
+  console.log('[greeting]', 'applying');
```
Comment (tayiorbeii) [inline lib/behavior/greeting/index.js]: do you want this log in here?

Comment (mbilokonsky) [inline lib/behavior/greeting/index.js]: Can pull that out of here and onboard user, but kinda liked the convention of having behaviors log that they're active. Can help to debug etc, and it's just one line in the heroku startup.

File: lib/behavior/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const onboard_new_user = require('./onboard_new_user');
+const greeting = require("./greeting");
+
+const enabledBehaviors = [
+  onboard_new_user, 
+  greeting  
+];
+
+const applyToBot = controler => enabledBehaviors.forEach(b => b(controller));
```
Comment (tayiorbeii) [inline lib/behavior/index.js]: `controller` typo

Comment (mbilokonsky) [inline lib/behavior/index.js]: how is that even working... thanks. I should turn on linting.

Comment (mbilokonsky) [inline lib/behavior/index.js]: oh, it's not working, I'm not calling that function. easy enough to fix.

File: lib/behavior/greeting/index.js (hunk 1 above)
Comment (joelhooks) [inline lib/behavior/greeting/index.js]: we can add winston or some such as a future improvement. Console "is the UI" though.

**Pull request merged**